core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -0,0 +1,362 @@
+package org.ensime.core
+
+import java.nio.file.Path
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props, Stash }
+import akka.event.LoggingReceive.withLabel
+import org.ensime.api.{ AnalyzerReadyEvent, ArchiveFile, AstAtPointReq, CompletionsReq, DocUriAtPointReq, DocUriForSymbolReq, EnsimeConfig, EnsimeProject, EnsimeProjectId, EnsimeServerError, ExpandSelectionReq, FullTypeCheckCompleteEvent, ImplicitInfoReq, InspectPackageByPathReq, InspectTypeAtPointReq, InspectTypeByNameReq, OffsetRange, PackageMemberCompletionReq, RawFile, RefactorReq, ReloadExistingFilesEvent, RemoveFileReq, RemoveFilesReq, RpcAnalyserRequest, RpcResponse, SourceFileInfo, StructureViewReq, SymbolAtPointReq, SymbolByNameReq, SymbolDesignationsReq, TypeAtPointReq, TypeByNameAtPointReq, TypeByNameReq, TypecheckAllReq, TypecheckFileReq, TypecheckFilesReq, TypecheckModule, UnloadAllReq, UnloadFileReq, UnloadModuleReq, UsesOfSymbolAtPointReq, VoidResponse }
+import org.ensime.config.RichEnsimeModule
+import org.ensime.util.FileUtils.toSourceFileInfo
+import org.ensime.util.ensimefile.richEnsimeFile
+import org.ensime.util.file.File
+
+import scala.collection.{ breakOut, mutable }
+
+trait ModuleFinder {
+  implicit val config: EnsimeConfig
+
+  def getModule(path: Path): Option[EnsimeProject] = {
+    config.projects.collectFirst {
+      case p if (p.sources.exists(f => path.toString.startsWith(f.toString))) =>
+        p
+    }
```

Comment:
I know I proposed the change to `collectFirst`, but now I see you rearranged code a bit and it is possible to use `find`.